### PR TITLE
LTD-5702: Update goods endpoint tests

### PR DESCRIPTION
**Pull request by saruniitr**
### Aim

[LTD-](https://uktrade.atlassian.net/browse/LTD-)

### CODE REVIEW

File: api/data_workspace/v2/tests/bdd/scenarios/destinations.feature [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
 Scenario: Check that the country code and type are included in the extract
-    Given a standard licence is created
-    When I fetch all destinations
-    Then the country code and type are included in the extract
+    Given a draft standard application with attributes:
+        | name | value                                |
+        | id   | 03fb08eb-1564-4b68-9336-3ca8906543f9 |
+    When the application is submitted
+    And the application is issued at 2024-11-22T13:35:15
+    Then the `destinations` table has the following rows:
```
Comment (kevincarrogan) [inline api/data_workspace/v2/tests/bdd/scenarios/destinations.feature:16]: I feel this test isn’t really telling a reader anything about how these values are getting set, they just sort of seem to appear from nowhere.

File: api/data_workspace/v2/tests/bdd/scenarios/goods.feature [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
-    When I fetch all goods
-    Then the non-draft good is included in the extract
-    And the draft good is not included in the extract
+    Given a draft standard application with attributes:
+        | name                    | value                                |
+        | id                      | 03fb08eb-1564-4b68-9336-3ca8906543f9 |
+        | good_on_application_id  | 94590c78-d0a9-406d-8fd3-b913bf5867a9 |
+    When the application is submitted
+    Then the `goods` table has the following rows:
+        | id                                   | application_id                        | quantity  | unit  | value    |
```
Comment (kevincarrogan) [inline api/data_workspace/v2/tests/bdd/scenarios/goods.feature:17]: This test feels the same as the one above where it’s not really describing how these values happen to be the values that end up in the table they just sort of exist for no apparent reason.

File: api/data_workspace/v2/tests/bdd/scenarios/destinations.feature (hunk 1 above)
Comment (saruniitr) [reply api/data_workspace/v2/tests/bdd/scenarios/destinations.feature:16]: ok, I was only trying to update the format, will improve the test.

File: api/data_workspace/v2/tests/bdd/scenarios/goods.feature (hunk 2 above)
Comment (saruniitr) [reply api/data_workspace/v2/tests/bdd/scenarios/goods.feature:19]: updated

File: api/data_workspace/v2/tests/bdd/scenarios/destinations.feature [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
 
 Scenario: Check that the country code and type are included in the extract
-    Given a standard licence is created
-    When I fetch all destinations
-    Then the country code and type are included in the extract
+    Given a draft standard application with attributes:
+        | name                | value                                |
+        | id                  | 03fb08eb-1564-4b68-9336-3ca8906543f9 |
+        | consignee_country   | AU                                   |
+        | end_user_country    | NZ                                   |
```
Comment (kevincarrogan) [inline api/data_workspace/v2/tests/bdd/scenarios/destinations.feature:14]: This still feels like we aren't writing this for our users and that we're sort of writing stuff to make our tests work.

We can leave this for future but I think something that would have described what the exporter does e.g.

```
Given a draft standard application
And a consignee added to the application in Australia
And an end user added to the application of New Zealand
```

**Review by kevincarrogan — APPROVED** [review]

**Review by hnryjmes — APPROVED** [review]

**Pull request merged**